testng-core/src/test/kotlin/test/hook/samples/HookSuccessSample.kt

```diff
@@ -0,0 +1,31 @@
+package test.hook.samples
+
+import test.hook.*
+
+import java.util.UUID
+import org.testng.IHookCallBack
+import org.testng.IHookable
+import org.testng.ITestResult
+import org.testng.Reporter
+import org.testng.annotations.DataProvider
+import org.testng.annotations.Test
+import test.TestData
+
+class HookSuccessSample : IHookable {
+
+    override fun run(callBack: IHookCallBack, testResult: ITestResult) {
+        testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
+        testResult.setAttribute(HOOK_METHOD_PARAMS_ATTRIBUTE, callBack.parameters)
+        callBack.runTestMethod(testResult)
+    }
+
+    @DataProvider
+    fun dp(): TestData = arrayOf(
+        arrayOf(UUID.randomUUID())
```

Comment:
Use a fixed uuid in order to avoid "random" failure (we can imagine that the generate may have a issue in some specific cases)

Reply:
Sure. I can do that. But can you let me know as  when would there be a use case wherein UUID generation itself fails?  This is just for my learning sake.